### PR TITLE
Sitegen: Refactor Error State

### PR DESCRIPTION
**Pull request by arunshenoy99**
## Type of Change

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [x] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments
This refactors the error state to make it more predictable, readable, and also fixes multiple bugs found within it.
1. Rewrite the site meta generation logic to be more predictable.
2. Avoid making step-specific API calls when in an error state.
3. Remove circular dependencies by ensuring that we disable the error state only if it was enabled initially.
4. Remove the error state on refresh: https://github.com/newfold-labs/wp-module-onboarding-data/pull/66
5. Utilize failure functions instead of directly modifying the state.

### CODE REVIEW

**Review by officiallygod — APPROVED** [review]
Looks good!

**Comment by girish-lokapure** [comment]
On the site-logo page, If you skip the upload the APIs are triggered twice.

**Comment by girish-lokapure** [comment]
On the site-logo page, if you choose to skip the upload, all APIs are triggered twice. 

This issue may occur because while on this step, the APIs are still pending, and if you skip the step, they are triggered again.

**Comment by arunshenoy99** [comment]
> On the site-logo page, if you choose to skip the upload, all APIs are triggered twice.
> 
> This issue may occur because while on this step, the APIs are still pending, and if you skip the step, they are triggered again.

Fixed, thanks @girish-lokapure @ajayadav09. Please let me know if you find any more issues.

File: src/OnboardingSPA/components/NewfoldInterfaceSkeleton/SiteGen/index.js [hunk 1]
```diff
@@ -172,25 +181,69 @@ const SiteGen = () => {
...
+			// Increase count after site meta calls to ensure systematic call of homepages
+			currentData.sitegen.siteGenMetaStatus.totalCount += 1;
+			// Get the homepages and set that in flow
+			const response = await getHomepages(
+				currentData.sitegen.siteDetails.prompt
+			);
+
+			if ( response.body ) {
+				currentData.sitegen.homepages.data = response.body;
+				currentData.sitegen.siteGenMetaStatus.currentStatus += 1;
```
Comment (ajayadav09) [inline src/OnboardingSPA/components/NewfoldInterfaceSkeleton/SiteGen/index.js:231]: Screenshot 2024-04-05 at 11 10 23 PM
Screenshot 2024-04-05 at 11 24 52 PM

i failed the homepage API with mock error. 
When the home page API fails in the background on the experience step and the user has selected one of the options from the survey question. The page doesn't move to the next step and the user is not aware of any errors, the user will be stuck on that step.

**Review by ajayadav09 — COMMENTED** [review]
Looks good. It'll be good in case to show the error UI when the home page api fails on the experience step. Have added screenshot in the comments.

**Comment by ajayadav09** [comment]
Looks Good! Thanks @arunshenoy99

**Review by ajayadav09 — APPROVED** [review]
Tested the home page error status. Working as expected

**Review by officiallygod — APPROVED** [review]
Looks Good!

**Pull request merged**